lib/types/bcachefs.nix

```diff
@@ -0,0 +1,124 @@
+{ config, options, diskoLib, lib, rootMountPoint, parent, device, ... }:
+{
+  options = {
+    type = lib.mkOption {
+      type = lib.types.enum [ "bcachefs" ];
+      internal = true;
+      description = "Type";
+    };
+    device = lib.mkOption {
+      type = lib.types.str;
+      default = device;
+      description = "Device to use";
+    };
+    extraArgs = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      description = "Extra arguments";
+    };
+    mountOptions = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ "defaults" ];
+      description = "A list of options to pass to mount.";
+    };
+    subvolumes = lib.mkOption {
+      type = lib.types.attrsOf (lib.types.submodule ({ config, ... }: {
+        options = {
+          name = lib.mkOption {
+            type = lib.types.str;
+            default = config._module.args.name;
+            description = "Name of the Bcachefs subvolume.";
+          };
+          type = lib.mkOption {
+            type = lib.types.enum [ "bcachefs_subvol" ];
+            default = "bcachefs_subvol";
+            internal = true;
+            description = "Type";
+          };
+          extraArgs = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [ ];
+            description = "Extra arguments";
+          };
+          mountpoint = lib.mkOption {
+            type = lib.types.nullOr diskoLib.optionTypes.absolute-pathname;
+            default = null;
+            description = "Location to mount the subvolume to.";
```

Comment:
if mountpoint has no effect, why not remove it?

Reply:
yep, I think it should just be the name